src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,110 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int CAPACITY = 10;
+    private Object[] elements;
+    private int count;
+
+    public ArrayList() {
+        elements = new Object[CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (isFull()) {
+            grow();
+        }
+        elements[count++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndexToAdd(index);
+        if (isFull()) {
+            grow();
+        }
+        System.arraycopy(elements, index, elements, index + 1, size() - index);
```

Comment:
you can use your variable count (better size) everywhere and not calling size() method. Check all places